nibabel/streamlines/tests/test_tractogram.py

```diff
@@ -233,6 +248,20 @@ def test_getitem(self):
             assert_arrays_equal(sdict[-1][k], v[-1])
             assert_arrays_equal(sdict[[0, -1]][k], v[[0, -1]])
 
+    def test_extend(self):
+        total_nb_rows = DATA['tractogram'].streamlines.total_nb_rows
+        sdict = PerArraySequenceDict(total_nb_rows, DATA['data_per_point'])
+        sdict2 = PerArraySequenceDict(total_nb_rows, DATA['data_per_point'])
+
+        sdict.extend(sdict2)
+        assert_equal(len(sdict), len(sdict2))
+        for k, v in DATA['tractogram'].data_per_point.items():
+            assert_arrays_equal(sdict[k][:len(DATA['tractogram'])], v)
+            assert_arrays_equal(sdict[k][len(DATA['tractogram']):], v)
+
+        # Test incompatible PerArrayDicts.
+        assert_raises(ValueError, sdict.extend, PerArraySequenceDict())
```

Comment:
Check for situation where there are extra keys in one or other?   Other than empty case here?